lib/apress/selenium_eti/spec/company_site/eti/product_creation_spec.rb

```diff
@@ -2,107 +2,98 @@
+          expect(@cs_eti_table_products.public_state(@product)).to eq(:unpublished)
         else
           raise ArgumentError, 'locator not found'
         end
       end
-
-      after(:all) { @cs_eti_page.delete_product(@name) }
     end
 
     context 'когда товар с рубрикой' do
       before(:all) do
-        @name = Faker::Number.number(digits: 5).to_s
-        @cs_eti_page.add_product
-        @cs_eti_page.set_name(@name)
-        @cs_eti_page.wait_saving
-        @cs_eti_page.set_rubric(CONFIG['eti']['rubric'])
-        @cs_eti_page.wait_until { @cs_eti_page.first_product_status_element.attribute('title') == 'Опубликованные на портале' }
-        @cs_eti_page.refresh
-        @cs_eti_page.search_product(@name)
-      end
-
-      it 'введенное имя отображается' do
-        expect(@cs_eti_page.product_name?(@name)).to be true
-      end
-
-      it 'рубрика привязана' do
-        expect(@cs_eti_page.product_rubric_tree(@name)).to include CONFIG['eti']['rubric']
+        @name = Faker::Number.leading_zero_number(digits: 5).to_s
+        @cs_eti_table_products.add_product(name: @name, rubric: CONFIG['eti']['rubric'])
+        @cs_eti_header.search_product(@name, exact: true)
+        @product = @cs_eti_table_products.product(name: @name)
       end
 
-      it 'товар опубликован' do
-        expect(@cs_eti_page.product_published?(@name)).to be true
-      end
+      after(:all) { @cs_eti_table_products.delete_product(@product) }
 
-      after(:all) { @cs_eti_page.delete_product(@name) }
+      it('введенное имя отображается') { expect(@cs_eti_table_products.name(@product)).to eq(@name) }
+      it('рубрика привязана') { expect(@cs_eti_table_products.rubric(@product)).to include(CONFIG['eti']['rubric']) }
+      it('товар опубликован') { expect(@cs_eti_table_products.public_state(@product)).to eq(:published) }
     end
 
     context 'когда копируем товар' do
       before(:all) do
-        @product = {
-          name: Faker::Number.leading_zero_number(digits: 5),
+        @fields = {
+          name: Faker::Number.leading_zero_number(digits: 10).to_s,
           rubric: CONFIG['eti']['rubric'],
-          exists: CONFIG['eti']['exists']['in stock'],
-          short_description: CONFIG['product_creation']['short_description']['valid'],
-          description: 'description',
-          price_from_to: {from: Faker::Number.number(digits: 3), to: Faker::Number.number(digits: 4)},
-          wholesale_price: {wholesale_price: Faker::Number.number(digits: 2), wholesale_number: Faker::Number.number(digits: 2)}
-        }
-
-        @portal_traits = {
-          trait_1: CONFIG['eti']['portal_traits']['trait_value_1'],
-          trait_2: CONFIG['eti']['portal_traits']['trait_value_2']
+          traits: {
+            CONFIG['eti']['portal_traits']['trait_1'] => CONFIG['eti']['portal_traits']['trait_value_1'],
+            CONFIG['eti']['portal_traits']['trait_2'] => CONFIG['eti']['portal_traits']['trait_value_2']
```

Comment:
Style/TrailingCommaInHashLiteral: Put a comma after the last item of a multiline hash.